4.2.3: build fails

**Describe the bug**
Build fails.

**To Reproduce**
- cmake setup:
```
[tkloczko@ss-desktop x86_64-redhat-linux-gnu]$ cmake -L
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
-- Cache values
BUILD_EXAMPLES:BOOL=OFF
BUILD_SHARED_LIBS:BOOL=OFF
BUILD_TESTING:BOOL=ON
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_INSTALL_PREFIX:PATH=/usr
CPACK_COMMAND:FILEPATH=/usr/bin/cpack
CURL_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/curl-config
DISCORDRPC_ENABLE_CLANG_FORMAT:BOOL=OFF
ENABLE_DISCORD:BOOL=ON
ENABLE_DOCS:BOOL=ON
ENABLE_IO_THREAD:BOOL=ON
ENABLE_NLS:BOOL=ON
FMT_CUDA_TEST:BOOL=OFF
FMT_DEBUG_POSTFIX:STRING=d
FMT_DOC:BOOL=OFF
FMT_FUZZ:BOOL=OFF
FMT_INC_DIR:STRING=include
FMT_INSTALL:BOOL=OFF
FMT_MODULE:BOOL=OFF
FMT_OS:BOOL=ON
FMT_PEDANTIC:BOOL=OFF
FMT_TEST:BOOL=OFF
FMT_WERROR:BOOL=OFF
GLSLC:FILEPATH=/usr/bin/glslc
GZIP_BIN:FILEPATH=/usr/bin/gzip
OpenAL_DIR:PATH=/usr/lib64/cmake/OpenAL
RAPIDJSON:FILEPATH=/usr/include/rapidjson
RE2_BUILD_TESTING:BOOL=OFF
SDL2_DIR:PATH=/usr/lib64/cmake/SDL2
SQLITECPP_BUILD_EXAMPLES:BOOL=OFF
SQLITECPP_BUILD_TESTS:BOOL=OFF
SQLITECPP_INCLUDE_SCRIPT:BOOL=OFF
SQLITECPP_INTERNAL_SQLITE:BOOL=OFF
SQLITECPP_RUN_CPPCHECK:BOOL=OFF
SQLITECPP_RUN_CPPLINT:BOOL=OFF
SQLITECPP_RUN_DOXYGEN:BOOL=OFF
SQLITECPP_USE_ASAN:BOOL=OFF
SQLITECPP_USE_GCOV:BOOL=OFF
SQLITECPP_USE_STACK_PROTECTION:BOOL=ON
SQLITECPP_USE_STATIC_RUNTIME:BOOL=OFF
SQLITE_ENABLE_ASSERT_HANDLER:BOOL=OFF
SQLITE_ENABLE_COLUMN_METADATA:BOOL=OFF
SQLITE_HAS_CODEC:BOOL=OFF
SQLITE_OMIT_LOAD_EXTENSION:BOOL=OFF
SQLITE_USE_LEGACY_STRUCT:BOOL=OFF
USEPCRE:BOOL=OFF
USE_STATIC_CRT:BOOL=OFF
UTF8PROC_ENABLE_TESTING:BOOL=OFF
UTF8PROC_INSTALL:BOOL=OFF
WARNINGS_AS_ERRORS:BOOL=OFF
WZ_DISTRIBUTOR:STRING=G2V
WZ_ENABLE_BACKEND_VULKAN:BOOL=ON
WZ_ENABLE_WARNINGS:BOOL=OFF
WZ_ENABLE_WARNINGS_AS_ERRORS:BOOL=OFF
WZ_FINDSDL2_NOCONFIG:BOOL=OFF
sodium_INCLUDE_DIR:PATH=/usr/include
sodium_LIBRARIES:FILEPATH=/usr/lib64/libsodium.so
```
- gcc 11.2.1

**Screenshots or Videos**
```console
[ 22%] Building CXX object lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/gfx_api_vk.cpp.o
cd /home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/ivis_opengl && /usr/bin/g++ -DHAVE_CONFIG_H -DJSON_USE_IMPLICIT_CONVERSIONS=0 -DWZ_VULKAN_ENABLED -I/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu -I/home/tkloczko/rpmbuild/BUILD/warzone2100/. -I/home/tkloczko/rpmbuild/BUILD/warzone2100/3rdparty -I/home/tkloczko/rpmbuild/BUILD/warzone2100/3rdparty/date/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/home/tkloczko/rpmbuild/BUILD/warzone2100/3rdparty/glad/include -isystem /home/tkloczko/rpmbuild/BUILD/warzone2100/3rdparty/glm -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -fstack-protector-strong -fstack-clash-protection -DNDEBUG -fPIC -Wpedantic -Wall -Wextra -fno-common -fno-math-errno -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Winit-self -Woverloaded-virtual -Wstrict-null-sentinel -Wwrite-strings -Wpointer-arith -Wmissing-noreturn -Wnon-virtual-dtor -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca-larger-than=1024 -Wrestrict -Wfloat-conversion -Wformat-security -Wstringop-truncation -Wlogical-op -Wuninitialized -Wno-float-conversion -Wno-sign-compare -Wno-unused-parameter -Wno-format-truncation -std=c++11 -MD -MT lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/gfx_api_vk.cpp.o -MF CMakeFiles/ivis-opengl.dir/gfx_api_vk.cpp.o.d -o CMakeFiles/ivis-opengl.dir/gfx_api_vk.cpp.o -c /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp: In member function ‘void BlockBufferAllocator::allocateNewBlock(uint32_t)’:
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:480:17: error: ‘throwResultException’ was not declared in this scope; did you mean ‘vk::{anonymous}::throwResultException’?
  480 |                 throwResultException( result, "vmaCreateBuffer" );
      |                 ^~~~~~~~~~~~~~~~~~~~
      |                 vk::{anonymous}::throwResultException
In file included from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.h:38,
                 from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:42:
/usr/include/vulkan/vulkan.hpp:6315:23: note: ‘vk::{anonymous}::throwResultException’ declared here
 6315 |     [[noreturn]] void throwResultException( Result result, char const * message )
      |                       ^~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp: In constructor ‘VkPSO::VkPSO(vk::Device, const vk::PhysicalDeviceLimits&, const gfxapi_PipelineCreateInfo&, vk::RenderPass, const std::shared_ptr<VkhRenderPassCompat>&, vk::SampleCountFlagBits, const vk::DispatchLoaderDynamic&)’:
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:1307:25: error: ‘throwResultException’ was not declared in this scope; did you mean ‘vk::{anonymous}::throwResultException’?
 1307 |                         throwResultException(result.result, "createGraphicsPipeline");
      |                         ^~~~~~~~~~~~~~~~~~~~
      |                         vk::{anonymous}::throwResultException
In file included from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.h:38,
                 from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:42:
/usr/include/vulkan/vulkan.hpp:6315:23: note: ‘vk::{anonymous}::throwResultException’ declared here
 6315 |     [[noreturn]] void throwResultException( Result result, char const * message )
      |                       ^~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp: In member function ‘void VkBuf::allocateBufferObject(const size_t&)’:
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:1371:17: error: ‘throwResultException’ was not declared in this scope; did you mean ‘vk::{anonymous}::throwResultException’?
 1371 |                 throwResultException( result, "vmaCreateBuffer" );
      |                 ^~~~~~~~~~~~~~~~~~~~
      |                 vk::{anonymous}::throwResultException
In file included from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.h:38,
                 from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:42:
/usr/include/vulkan/vulkan.hpp:6315:23: note: ‘vk::{anonymous}::throwResultException’ declared here
 6315 |     [[noreturn]] void throwResultException( Result result, char const * message )
      |                       ^~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp: In constructor ‘VkTexture::VkTexture(const VkRoot&, const size_t&, const size_t&, const size_t&, const vk::Format&, const string&)’:
/home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:1489:17: error: ‘throwResultException’ was not declared in this scope; did you mean ‘vk::{anonymous}::throwResultException’?
 1489 |                 throwResultException( result, "vmaCreateImage" );
      |                 ^~~~~~~~~~~~~~~~~~~~
      |                 vk::{anonymous}::throwResultException
In file included from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.h:38,
                 from /home/tkloczko/rpmbuild/BUILD/warzone2100/lib/ivis_opengl/gfx_api_vk.cpp:42:
/usr/include/vulkan/vulkan.hpp:6315:23: note: ‘vk::{anonymous}::throwResultException’ declared here
 6315 |     [[noreturn]] void throwResultException( Result result, char const * message )
      |                       ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/build.make:149: lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/gfx_api_vk.cpp.o] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu'
```
**Your System:**
 - OS: Linux/x86_64
 - Game version: 4.2.3

**Additional context**
N/A

--- a/lib/ivis_opengl/gfx_api_vk.cpp
+++ b/lib/ivis_opengl/gfx_api_vk.cpp
@@ -477,7 +477,7 @@ void BlockBufferAllocator::allocateNewBlock(uint32_t minimumSize)
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateBuffer" );
+		vk::throwResultException( result, "vmaCreateBuffer" );
 	}
 
 	if (autoMap)
@@ -1304,7 +1304,7 @@ VkPSO::VkPSO(vk::Device _dev,
 			object = std::move(result.value);
 			break;
 		default:
-			throwResultException(result.result, "createGraphicsPipeline");
+			vk::throwResultException(result.result, "createGraphicsPipeline");
 	}
 }
 
@@ -1368,7 +1368,7 @@ void VkBuf::allocateBufferObject(const std::size_t& size)
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateBuffer" );
+		vk::throwResultException( result, "vmaCreateBuffer" );
 	}
 
 	buffer_size = size;
@@ -1486,7 +1486,7 @@ VkTexture::VkTexture(const VkRoot& root, const std::size_t& mipmap_count, const 
 	if (result != vk::Result::eSuccess)
 	{
 		// Failed to allocate memory!
-		throwResultException( result, "vmaCreateImage" );
+		vk::throwResultException( result, "vmaCreateImage" );
 	}
 
 	const auto imageViewCreateInfo = vk::ImageViewCreateInfo()

